lua/core/lualine/components.lua

```diff
@@ -118,21 +118,20 @@ return {
     function()
       local current_line = vim.fn.line "."
       local total_lines = vim.fn.line "$"
-      local default_chars = { "__", "▁▁", "▂▂", "▃▃", "▄▄", "▅▅", "▆▆", "▇▇", "██" }
-      local chars = default_chars
-      local index = 1
+      local chars = { "__", "▁▁", "▂▂", "▃▃", "▄▄", "▅▅", "▆▆", "▇▇", "██" }
+      local scroll_index = 1
       if current_line == 1 then
-        index = 1
+        scroll_index = 1
       elseif current_line == total_lines then
-        index = #chars
+        scroll_index = #chars
       else
         local line_no_fraction = vim.fn.floor(current_line) / vim.fn.floor(total_lines)
```

Comment:
I have some doubts about this block of code.
My first concern is that the two conditions above should not be necessary.
Then `local line_no_fraction = vim.fn.floor(current_line) / vim.fn.floor(total_lines)` sound weird to me, does `vim.fn.floor` works the same way as [math.floor](https://devdocs.io/lua~5.4/index#pdf-math.floor) ? also, this function is being passed line numbers, aren't they already integers ? Finally, line_no_fraction will be a fraction if we end up dividing two integers, am I missing smth ?

Reply:
You are not missing anything :)